src/osd/ReplicatedPG.cc

```diff
@@ -9893,6 +9893,8 @@ void ReplicatedPG::mark_all_unfound_lost(
       [=]() {
 	requeue_ops(waiting_for_all_missing);
 	waiting_for_all_missing.clear();
+	requeue_object_waiters(waiting_for_unreadable_object);
+	// XXX: What about waiting_for_degraded_object, waiting_for_blocked_object?
```

Comment:
I think waiting_for_degraded_object must be empty here.  Same with waiting_for_blocked_object.  You could asserrt to that effect.

Also, I think you should be only requeueing ops for the objects marked lost.